keel-deliveryconfig-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/deliveryconfig/DeliveryEnvironment.kt

```diff
@@ -0,0 +1,36 @@
+package com.netflix.spinnaker.keel.api.deliveryconfig
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
+import com.netflix.spinnaker.keel.api.ApiVersion
+import com.netflix.spinnaker.keel.api.ResourceMetadata
+
+data class DeliveryEnvironment(
+  val name: String,
+  val packageRef: PackageRef,
+  val targets: List<Target>
+)
+
+interface SubResource {
```

Comment:
the interface + base class + typealiases in here exist because at one point I thought I would have additional attributes on the different types, but in hindsight I think those don't make sense with the model of treating things directly as resources (I'd pull any attributes up to the DeliveryEnvironment level)

Reply:
so I can and probably will eventually just make this a single type, but I'd prefer to leave it for now and address it in the follow up change where we start exploring a better way to do type referencing

Reply:
I find this very confusing